src/main/java/org/jabref/logic/util/io/FileNameUniqueness.java

```diff
@@ -41,4 +45,90 @@ public static String getNonOverWritingFileName(Path targetDirectory, String file
 
         return newFileName;
     }
+
+    /**
+     *
+     * @param directory The directory which saves the files (.pdf, for example)
+     * @param fileName Suggest name for the newly downloaded file
+     * @param dialogService To display the error and success message
+     * @return true when the content of files is duplicate and successfully delete the newly downloaded file ,
+     *         false when it is not a duplicate file or fail to delete the duplicate file
+     * @throws IOException Fail when the file is not exist or something wrong when reading the file
+     */
+    public static boolean isDuplicatedFile(Path directory, String fileName, DialogService dialogService) throws IOException {
+
+        Optional<String> extensionOptional = FileUtil.getFileExtension(fileName);
+        String extensionSuffix;
+        String fileNameWithoutDuplicated;
+
+        if (extensionOptional.isPresent()) {
+            extensionSuffix = '.' + extensionOptional.get();
+            fileNameWithoutDuplicated = eraseDuplicateMarks(fileName.substring(0, fileName.lastIndexOf('.')));
+        } else {
+            extensionSuffix = "";
+            fileNameWithoutDuplicated = eraseDuplicateMarks(fileName);
+        }
```

Comment:
Please use the available methods in `FileUtil` and the power of Java optionals

```suggestion
        String fileNameWithoutDuplicated = eraseDuplicateMarks(FileUtil.getBaseName(fileName));
        String extensionSuffix = FileUtil.getFileExtension(fileName).getOrElse("");
```